core/src/main/java/greencity/mapping/AddEcoNewsDtoRequestMapper.java

```diff
@@ -41,16 +52,26 @@ public AddEcoNewsDtoRequestMapper(LanguageRepository languageRepository) {
     protected EcoNews convert(AddEcoNewsDtoRequest addEcoNewsDtoRequest) {
         EcoNews ecoNews = EcoNews.builder()
             .creationDate(ZonedDateTime.now())
-            .text(addEcoNewsDtoRequest.getText())
+            .author(userRepo.findById(addEcoNewsDtoRequest.getAuthor().getId()).orElseThrow(
```

Comment:
Converter should not have logic related to get info from DB or some business logic. This logic should live in repos or services. Converters only responsibility to map data.